### PR TITLE
feat(sbb-action-group): initial implementation

### LINKED ISSUE #1368: story(sbb-action-group): initial implementation
### Feature Description

We want to provide the sbb-action-group component.

[Figma Spec](https://www.figma.com/file/9r6xSfNmEfCFxl1yFYedrj/Lyne-Components?node-id=7965%3A296702)

**Design Spec**
- The `<sbb-action-group>` should support both `<sbb-button>` and `<sbb-link>`

**Technical Spec**
- Use CSS Flexbox (with `gap`, etc.)
- Provide property `orientation: 'horizontal' | 'vertical' = 'horizontal'`
- Provide property `horizontalFrom?: 'zero' | 'micro' | 'small' | 'medium' | 'large' | 'wide' | 'ultra'`, which overrides the `orientation` property
- Provide property `align: 'start' | 'center' | 'end' = 'start'` (To discuss; Add 'stretch' as a differentiation to 'center')
- Support attribute `align-self="start"` `align-self="end"` on `<sbb-button>` or `<sbb-link>` to allow a single instance to be moved contrary to the group alignment

```html
<sbb-action-group>
  <sbb-button variant="secondary">Action 1</sbb-button>
  <sbb-button>Action 2</sbb-button>
</sbb-action-group>
```

```html
<sbb-action-group align="end">
  <sbb-button align-self="start" variant="secondary">Action 1</sbb-button>
  <sbb-button variant="secondary">Action 2</sbb-button>
  <sbb-button>Action 3</sbb-button>
</sbb-action-group>
```

```html
<sbb-action-group orientation="vertical">
  <sbb-button variant="secondary">Action 1</sbb-button>
  <sbb-button variant="secondary">Action 2</sbb-button>
  <sbb-button>Action 3</sbb-button>
</sbb-action-group>
```

### Definition of Done

 - [ ]  Component is implemented
 - [ ]  Documentation is provided
 - [ ]  Tests are implemented
 - [ ]  E2E Tests are implemented
 - [ ]  Storybook stories are implemented
 - [ ]  Navigation via keyboard is tested
 - [ ]  Screen reader output is tested
 - [ ]  High-contrast is tested
 - [ ]  Remaining accessibility is tested
 - [ ]  UX approved

### Definition of Done

- [ ] Component is implemented
- [ ] Documentation is provided
- [ ] Tests are implemented
- [ ] E2E Tests are implemented
- [ ] Storybook stories are implemented
- [ ] Navigation via keyboard is tested
- [ ] Screen reader output is tested
- [ ] High-contrast is tested
- [ ] Remaining accessibility is tested
- [ ] UX approved

### PR DESCRIPTION
**Pull request by simone-sabato**
## Preflight Checklist

- [x] I have read the [Contributing Guidelines](https://github.com/lyne-design-system/lyne-components/blob/master/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/lyne-design-system/lyne-components/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [pull request tracker](https://github.com/lyne-design-system/lyne-components/pulls) for a Pull Request (PR) that matches the one I want to submit, without success.

## Issue

This PR Closes #1368

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)

See [Review Guidelines](../REVIEW.md) for more information what is checked during review process.

## Changes

Changes in this pull request:

- implement action group component

## Browsers

I tested the build on the following browsers:

- [ ] Firefox Desktop
- [ ] Chrome Desktop
- [ ] Edge Desktop
- [ ] Safari Desktop
- [ ] Chrome Mobile
- [ ] Safari Mobile

## Screen readers

I tested the build on the following browsers:

- [ ] JAWS Firefox Desktop
- [ ] JAWS Chrome Desktop
- [ ] NVDA Firefox Desktop
- [ ] NVDA Chrome Desktop
- [ ] VoiceOver Safari Desktop
- [ ] VoiceOver Chrome Desktop
- [ ] VoiceOver Safari Mobile
- [ ] Android Accessibility Suite Chrome Mobile

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/components/sbb-action-group/sbb-action-group.scss [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+  }
+
+  .action-group--align-end {
+    --sbb-action-group-align-items: flex-end;
+  }
+
+  .action-group {
+    align-items: var(--sbb-action-group-align-items);
+  }
+}
```
Comment (dauriamarco) [inline src/components/sbb-action-group/sbb-action-group.scss:25]:
```suggestion
$breakpoints: 'zero', 'micro', 'small', 'medium', 'large', 'wide', 'ultra';

$vertical-orientations: (
  start: flex-start,
  center: center,
  stretch: stretch,
  end: flex-end,
);

$horizontal-orientations: (
  start: flex-start,
  center: center,
  stretch: space-between,
  end: flex-end,
);

@mixin vertical-action-group {
  @each $key, $value in $vertical-orientations {
    .action-group--align-#{$key} {
      --sbb-action-group-align-items: #{$value};
    }
  }

  .action-group {
    align-items: var(--sbb-action-group-align-items);
  }
}
```

File: src/components/sbb-action-group/sbb-action-group.scss [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+    --sbb-action-group-justify-content: center;
+  }
+
+  .action-group--align-stretch {
+    --sbb-action-group-justify-content: space-between;
+  }
+
+  .action-group--align-end {
+    --sbb-action-group-justify-content: flex-end;
+  }
```
Comment (dauriamarco) [inline src/components/sbb-action-group/sbb-action-group.scss:74]:
```suggestion
@each $key, $value in $horizontal-orientations {
    .action-group--align-#{$key} {
      --sbb-action-group-justify-content: #{$value};
    }
  }
```

File: src/components/sbb-action-group/sbb-action-group.scss [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+    @include vertical-action-group-element-align-reset;
+  }
+}
+
+:host([orientation='horizontal'][horizontal-from='ultra']),
+:host([orientation='vertical'][horizontal-from='ultra']) {
+  @include mq($to: ultra) {
+    @include vertical-action-group-element-align-reset;
+  }
+}
```
Comment (dauriamarco) [inline src/components/sbb-action-group/sbb-action-group.scss:188]:
```suggestion
@each $breakpoint in $breakpoints {
  // horizontal-from overrides orientation vertical
  :host([orientation='vertical'][horizontal-from='#{$breakpoint}']) {
    --sbb-action-group-orientation: row;

    @each $key, $value in $horizontal-orientations {
      .action-group--align-#{$key} {
        --sbb-action-group-justify-content: #{$value};
        --sbb-action-group-align-items: unset;
      }

      .action-group {
        justify-content: var(--sbb-action-group-justify-content);
      }

      ::slotted([align-self='start']) {
        margin-inline-end: auto;
      }

      ::slotted([align-self='center']) {
        margin-inline: auto;
      }

      ::slotted([align-self='end']) {
        margin-inline-start: auto;
      }
    }
  }

  // horizontal-from overrides orientation horizontal
  :host([orientation='horizontal'][horizontal-from='#{$breakpoint}']),
  :host([orientation='vertical'][horizontal-from='#{$breakpoint}']) {
    @include mq($to: #{$breakpoint}) {
      @include vertical-action-group-element-align-reset;
    }
  }
}
```

**Review by dauriamarco — COMMENTED** [review]
In this context we find ourselves repeating many styles based on breakpoints and alignments. I think we could greatly simplify by widely using SASS maps and "at-rules" such as `@each`.

I'll try to come up with an alternative. 😄

**Review by dauriamarco — COMMENTED** [review]
Hi guys, can you update the PR checklists and the "Definition of Done" in the issue?

File: src/components/sbb-action-group/sbb-action-group.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+  /**
+   * Indicates the orientation of the components inside the `<sbb-action-group>`.
+   */
+  @Prop({ reflect: true }) public orientation: InterfaceSbbActionGroupAttributes['orientation'] =
+    'horizontal';
+
+  public render(): JSX.Element {
+    return (
+      <Host>
```
Comment (DavideMininni-Fincons) [inline src/components/sbb-action-group/sbb-action-group.tsx:32]: Tag can be removed if unused (no classes, no attributes..)

**Review by DavideMininni-Fincons — COMMENTED** [review]
Looks good, see comment

**Review by DavideMininni-Fincons — COMMENTED** [review]
Stories 
https://5ea7d9e48c451f00223b52ff-nbcabmszld.chromatic.com/?path=/story/components-layout-sbb-action-group--vertical-allocation-0-0-3-full-width
and
https://5ea7d9e48c451f00223b52ff-nbcabmszld.chromatic.com/?path=/story/components-layout-sbb-action-group--vertical-allocation-0-3-0-full-width
looks different from Figma: the link should move at the end/at center.

File: src/components/sbb-action-group/sbb-action-group.scss [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+.sbb-action-group {
+  display: flex;
+  flex-direction: var(--sbb-action-group-orientation);
+  gap: var(--sbb-spacing-fixed-4x);
+  justify-content: var(--sbb-action-group-justify-content);
+  align-items: var(--sbb-action-group-align-items);
+
+  ::slotted([align-self='start']) {
+    margin-inline-end: auto;
```
Comment (jeripeierSBB) [inline src/components/sbb-action-group/sbb-action-group.scss:63]: Why is not align-self used here?

File: src/components/sbb-action-group/sbb-action-group.scss [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    margin-inline: auto;
+  }
+
+  ::slotted([align-self='end']) {
+    margin-inline-start: auto;
+  }
+}
+
+::slotted(sbb-link) {
+  white-space: nowrap;
```
Comment (jeripeierSBB) [inline src/components/sbb-action-group/sbb-action-group.scss:76]: I'm not sure if this is too hacky, but maybe is ok for now.

File: src/components/sbb-action-group/sbb-action-group.stories.js [hunk 7]
```diff
@@ -0,0 +1,264 @@
+import { h } from 'jsx-dom';
```
Comment (jeripeierSBB) [inline src/components/sbb-action-group/sbb-action-group.stories.js:1]: Can you add the following story:
- a story which has set horizontal from to a breakpoint (e.g. medium). So we can track that with chromatic tests too.

Comment (jeripeierSBB) [inline src/components/sbb-action-group/sbb-action-group.stories.js:1]: Can you maybe add the control for the size property which passes its size into the sbb-button?

**Review by jeripeierSBB — CHANGES_REQUESTED** [review]
Looks good, some story additions requested

**Comment by jeripeierSBB** [comment]
> Stories https://5ea7d9e48c451f00223b52ff-nbcabmszld.chromatic.com/?path=/story/components-layout-sbb-action-group--vertical-allocation-0-0-3-full-width and https://5ea7d9e48c451f00223b52ff-nbcabmszld.chromatic.com/?path=/story/components-layout-sbb-action-group--vertical-allocation-0-3-0-full-width looks different from Figma: the link should move at the end/at center.

I think it's fixed now

File: src/components/sbb-action-group/sbb-action-group.scss (hunk 5 above)
Comment (DavideMininni-Fincons) [reply src/components/sbb-action-group/sbb-action-group.scss:63]: From what I remember, align-self moves the elements in the cross axis; in this case we want to move them along the main axis.

**Review by jeripeierSBB — APPROVED** [review]
LGTM

**Pull request merged**